src/types.jl

```diff
@@ -454,17 +532,16 @@ function ReplicaSystem(;
     
     K = typeof(k_converted)
 
-    # TODO:  CellListNeighborFinder is mutable so can't be passed to all replicas.
-    replicas = Tuple(System{D, G, T, A, AD, PI, SI, GI, C, V, B, NF, L, F, E, K}(
+    replicas = Tuple(System{D, G, T, CU, A, AD, PI, SI, GI, C, V, B, NF, typeof(replica_loggers[i]), F, E, K}(
             atoms, atoms_data, pairwise_inters, specific_inter_lists,
-            general_inters, copy(coords), copy(velocities), boundary, neighbor_finder,
-            loggers[i], force_units, energy_units, k_converted) for i=1:n_replicas)
+            general_inters, copy(coords), copy(velocities), boundary, deepcopy(neighbor_finder),
```

Comment:
Right now a deepcopy of neighbor_finder is made in order to support `CellListNeighborFinder`. This can be made more efficient by defining a method of `Base.copy` for neighbor finders which passes the field that are not mutated by reference and then using this.

For example, for `DistanceNeighborFinder` it can be like:
```julia
Base.copy(nf::DistanceNeighborFinder{D, B, I}) = DistanceNeighborFinder{D, B, I}([getfield(x, k) for k ∈ fieldnames(T)]...)
``` 

whereas for `CellListNeighborFinder` we can have something like:
```julia
Base.copy(clnf::CellListMapNeighborFinder{N, T}) = CellListMapNeighborFinder{N, T}(
    clnf.nb_matrix, clnf.matrix_14, clnf.n_steps, clnf.dist_cutoff,
    deepcopy(clnf.cl), deepcopy(clnf.aux), deepcopy(clnf.neighbors_threaded)
)
```

Then any custom neighbor finder will need to define this method for it to be used in replica simulation.

Reply:
Not sure it's worth speeding this up as it is only called once, unless it is really slow.

Reply:
Yeah seems fair. I will keep it this way.